packages/framework/azure-client/src/interfaces.ts

```diff
@@ -35,17 +35,17 @@ export interface AzureClientProps {
  */
 export interface AzureConnectionConfig {
```

Comment:
Let's split this out into explicit types that will give us strong typing later.

```ts
export type AzureConnectionConfigType = "remote" | "local"
export interface AzureConnectionConfig {
  type: AzureConnectionConfigType;
}

export interface RemoteAzureConnectionConfig extends AzureConnectionConfig {
  type: "remote";
  tenantId: string;
  tokenProvider: ITokenProvider;
}

export interface LocalAzureConnectionConfig extends AzureConnectionConfig {
    type: "local";
    tokenProvider: ITokenProvider
}
```

Reply:
done 👍